lib/vector/diglib/frmt.c

```diff
@@ -46,7 +48,10 @@ int dig_read_frmt_ascii(FILE *dascii, struct Format_info *finfo)
             return -1;
         }
 
-        strcpy(buf1, buff);
+        len = G_strlcpy(buf1, buff, sizeof(buf1));
+        if (len >= sizeof(buf1)) {
+            G_fatal_error(_("Name <%s> is too long"), buff);
+        }
```

Comment:
This function's doc says it returns -1 on error, so it should not call fatal.

Couple lines above it gives warning on error and returns -1.

Reply:
done changes